include/irr/asset/IVirtualTexture.h

```diff
@@ -0,0 +1,750 @@
+                m_usamplers.views = ucount ? core::make_refctd_dynamic_array<decltype(SamplerArray::views)>(ucount) : nullptr;
+            }
+            {
+                uint32_t fi = 0u, ii = 0u, ui = 0u;
+
+                execPerFormat(
+                    [&fi, this](E_FORMAT _fmt, IVTResidentStorage* _storage) { (*m_fsamplers.views)[fi++] = _storage->createView(_fmt); },
+                    [&ii, this](E_FORMAT _fmt, IVTResidentStorage* _storage) { (*m_isamplers.views)[ii++] = _storage->createView(_fmt); },
+                    [&ui, this](E_FORMAT _fmt, IVTResidentStorage* _storage) { (*m_usamplers.views)[ui++] = _storage->createView(_fmt); }
+                );
+            }
+        }
+    }
+
+public:
+    IVirtualTexture(
+        uint32_t _pgTabSzxy_log2 = 8u,
+        uint32_t _pgTabLayers = 256u,
+        uint32_t _pgSzxy_log2 = 7u,
+        uint32_t _tilePadding = 9u,
+        bool _initSharedResources = true
+    ) :
+        m_pgSzxy(1u<<_pgSzxy_log2), m_pgSzxy_log2(_pgSzxy_log2), m_pgtabSzxy_log2(_pgTabSzxy_log2), m_tilePadding(_tilePadding),
+        m_pageTableLayerAllocators(core::make_refctd_dynamic_array<decltype(m_pageTableLayerAllocators)>(_pgTabLayers)),
+        m_layerToViewIndexMapping(core::make_refctd_dynamic_array<decltype(m_layerToViewIndexMapping)>(_pgTabLayers,~0u))
+    {
+        if (_initSharedResources)
+        {
+            uint32_t pgtabSzSqr = (1u<<_pgTabSzxy_log2);
+            pgtabSzSqr *= pgtabSzSqr;
+            const size_t spacePerAllocator = pg_tab_addr_alctr_t::reserved_size(pgtabSzSqr, pgtabSzSqr, 1u);
+            m_pgTabAddrAlctr_reservedSpc = reinterpret_cast<uint8_t*>( _IRR_ALIGNED_MALLOC(spacePerAllocator*_pgTabLayers, _IRR_SIMD_ALIGNMENT) );
+            for (uint32_t i = 0u; i < _pgTabLayers; ++i)
+            {
+                auto& alctr = (*m_pageTableLayerAllocators)[i];
+                alctr = pg_tab_addr_alctr_t(m_pgTabAddrAlctr_reservedSpc+i*spacePerAllocator, 0u, 0u, pgtabSzSqr, pgtabSzSqr, 1u);
+            }
+        }
+
+        if (_initSharedResources)
+        {
+            decltype(m_freePageTableLayerIDs)::container_type stackBackend;
+            stackBackend.resize(_pgTabLayers);
+            for (uint32_t i = 0u; i < _pgTabLayers; ++i)
+                stackBackend[i] = _pgTabLayers-1u-i;
+            m_freePageTableLayerIDs = decltype(m_freePageTableLayerIDs)(std::move(stackBackend));
+        }
+        {
+            const uint32_t pgSzLog2 = getPageExtent_log2();
+            bool ok = SMiptailPacker::computeMiptailOffsets(m_miptailOffsets, pgSzLog2, m_tilePadding);
+            assert(ok);
+        }
+    }
+
+    virtual STextureData pack(const image_t* _img, const IImage::SSubresourceRange& _subres, ISampler::E_TEXTURE_CLAMP _wrapu, ISampler::E_TEXTURE_CLAMP _wrapv, ISampler::E_TEXTURE_BORDER_COLOR _borderColor) = 0;
+
+    virtual core::smart_refctd_ptr<image_view_t> createPageTableView() const = 0;
```

Comment:
this should be a `get` and should return a raw-pointer (cause the return value should be cached for all time) see my other comments